### PR TITLE
fixing code unreachable

### LINKED ISSUE #102: Coverity - code unreachable
Event unreachable:	Since the loop increment "el = el->NextSiblingElement..." is unreachable, the loop body will never execute more than once.
https://github.com/ros/rospack/blob/3358d21fbeb5355a459e94354c2aa95e0a376e57/src/rospack.cpp#L182

### PR DESCRIPTION
**Pull request by danielwangksu**
fix https://github.com/ros/rospack/issues/102
Signed-off-by: Daniel Wang <daniel.wang@canonical.com>

### CODE REVIEW

File: src/rospack.cpp [hunk 1]
```diff
@@ -179,11 +179,8 @@ class Stackage
       assert(manifest_loaded_);
       // get name from package.xml instead of folder name
       tinyxml2::XMLElement* root = get_manifest_root(this);
-      for(tinyxml2::XMLElement* el = root->FirstChildElement("name"); el; el = el->NextSiblingElement("name"))
-      {
-        name_ = el->GetText();
-        break;
-      }
+      if(root->FirstChildElement("name"))
+        name_ = root->GetText();
```
Comment (dirk-thomas) [inline src/rospack.cpp]: Wasn't the former logic setting `name_` to `root->FirstChildElement("name")->GetText()`?

**Review by dirk-thomas — COMMENTED** [review]
How have you tested the code change (before/after)?

File: src/rospack.cpp (hunk 1 above)
Comment (danielwangksu) [reply src/rospack.cpp]: You are right. Sorry missed it

File: src/rospack.cpp [hunk 2]
```diff
@@ -179,11 +179,8 @@ class Stackage
       assert(manifest_loaded_);
       // get name from package.xml instead of folder name
       tinyxml2::XMLElement* root = get_manifest_root(this);
-      for(tinyxml2::XMLElement* el = root->FirstChildElement("name"); el; el = el->NextSiblingElement("name"))
-      {
-        name_ = el->GetText();
-        break;
-      }
+      if(root->FirstChildElement("name"))
+        name_ = root->FirstChildElement("name")->GetText();
```
Comment (dirk-thomas) [inline src/rospack.cpp]: Maybe extract the duplicate call into a local variable: `tinyxml2::XMLElement* el = root->FirstChildElement("name");`?

Comment (danielwangksu) [reply src/rospack.cpp]: sure!

**Comment by dirk-thomas** [comment]
Thanks for the improvement.

**Pull request merged**